src/metadatacache/MetadataCache.ts

```diff
@@ -312,6 +312,19 @@ export class MetadataCache {
     return result
   }
 
+  /**
+   * Validate DDO.
+   * @param  {ddp} string DDO.
+   * @return {Promise<any>} Result (200 OK, 500 INVALID).
+   */
+  public async validate(ddo: DDO): Promise<any> {
+    return await this.fetch
+      .post(`${this.url}${apiPath}/validate`, JSON.stringify(ddo))
```

Comment:
The validate endpoint expects the metadata json not the whole DDO. Get the metadata service `ddo.service[0].attributes`

Reply:
`ddo.findServiceByType('metadata)`. Never assuming the metadata service is the first service is safer.

Reply:
thanks for the feedback .. working on it.